x-pack/plugins/infra/public/pages/metrics/metric_detail/metric_detail_page.tsx

```diff
@@ -53,15 +57,26 @@ export const MetricDetailPage = () => {
   });
 
   const breadcrumbOptions = parentBreadcrumbResolver.getBreadcrumbOptions(nodeType);
-  useMetricsBreadcrumbs([
-    {
-      ...breadcrumbOptions.link,
-      text: breadcrumbOptions.text,
-    },
-    {
-      text: name,
-    },
-  ]);
+  const breadcrumbs = useMemo(
+    () => [
+      {
+        ...breadcrumbOptions.link,
+        text: breadcrumbOptions.text,
+      },
+      {
+        text: name,
+      },
+    ],
+    [breadcrumbOptions.link, breadcrumbOptions.text, name]
+  );
+  useMetricsBreadcrumbs(breadcrumbs);
+  useEffect(() => {
+    if (serverless) {
+      // For deeper context breadcrumbs serveless provides its own breadcrumb service.
+      // docs.elastic.dev/kibana-dev-docs/serverless-project-navigation#breadcrumbs
+      serverless.setBreadcrumbs(breadcrumbs);
+    }
+  }, [serverless, breadcrumbs]);
```

Comment:
I think this could be exported to its own generic custom hook. It looks like a boilerplate we need to do in other to add breadcrumbs to the pages we have.

Reply:
Yeah, good point. I'll see what I can do.